conan/tools/cmake/toolchain/blocks.py

```diff
@@ -1003,6 +1003,46 @@ def context(self):
                 "winsdk_version": winsdk_version,
                 "gen_platform_sdk_version": gen_platform_sdk_version}
 
+class ExtraVariablesBlock(Block):
+    template = textwrap.dedent(r"""
+        {% if extra_variables %}
+        {% for key, value in extra_variables.items() %}
+        set({{ key }} {{ value }})
+        {% endfor %}
+        {% endif %}
+    """)
+
+    CMAKE_CACHE_TYPES = ["BOOL","FILEPATH", "PATH", "STRING", "INTERNAL"]
+
+    def get_exact_type(self, key, value):
+        if type(value) == str:
+            return f"\"{value}\""
+        elif type(value) == int or type(value) == float:
+            return value
+        elif type(value) == dict:
+            var_value = self.get_exact_type(key, value.get("value"))
+            is_cache = value.get("cache")
+            if is_cache == True or str(is_cache).lower() == 'true':
```

Comment:
```suggestion
            if is_cache:
```
The value coming from the dict should already be a boolean or ``None`` if not defined. You can assert if want to make sure, rather than accept other types and convert like ``str``

Reply:
Oh, I thought it could be nice to capture both True/False and 'true'/'false'.
If we only want to accept True/False pythonic booleans, it is easier.